engine/engine.go

```diff
@@ -97,7 +98,26 @@ func (game *State) PlayUp(p uint8, c Card) error {
 
 // Deal shuffles a deck of cards and deals it out to each player.
 func (game *State) Deal() error {
-	return errors.New("not implemented")
+	deck := stdDeck()
+
+	shuffle(deck)
+
+	for i := 0; i < len(game.players); i++ {
+		player := &game.players[i]
```

Comment:
I'm a little confused by this because I thought players was an array of Player not *Player?

Reply:
I need to use player as a reference here instead of copying it. `game.Players` is an array of `Player`, you are correct, so if I want to pass a reference to a player I need to make one with `&`. 

The thing here that blew my mind a little is that when you make a variable to something, it also can be either a copy or a reference (I did not even think of that haha). I need to set this players hand below so I want to have the reference so that I can mutate it meaningfully.

Reply:
I could have set `game.player[i].hand` below and worked with game.player[i].hand but I think that would be less readable. And readability counts!

Reply:
Oh I see. I thought you could only use & to generate a pointer to something and that the variable you assign it to would also have to have a pointer type.